examples/key-recognition.js

```diff
@@ -0,0 +1,69 @@
+var x11 = require('..');
+
+// The list provided by GetKeyboardMapping has sublists where each position
+// represents the result for some key composition.
+keyComposition = [
+  'Key',
+  'shift + Key',
+  'modeSwitch + Key',
+  'modeSwitch + shift + Key',
+  'altGr + Key',
+  'altGr + shift + Key'
+]
+
+x11.createClient(function(err, display){
+  if (err) throw err
+
+  var min = display.min_keycode,  // TODO: explain min_keycode
+      max = display.max_keycode,  // TODO: explain max_keycode
+      chr2Data = [],              // allow us to find a char by the charcode
+      key2Data = [];              // associate chars to a keycode
+
+  // The keySyms is a hash of mnemonic char names, associated to an integer
+  // charcode and its description.
+  for (codeName in x11.keySyms) {
+    keyData = x11.keySyms[codeName];
+    chr2Data[keyData.code] = { codeName: codeName, description: keyData.description };
+  }
+
+  var X = display.client,
+      wid = X.AllocID(),                     // Get a free integer id to a new window.
+      root = display.screen[0].root,         // The mother window. Like your window manager.
+      evKeyPress = x11.eventMask.KeyPress,   // Allow to filter for KeyPress events.
+      white = display.screen[0].white_pixel; // Will paint the window.
+
+  // Get the local key mapping to build key2Data.
+  X.GetKeyboardMapping(min, max-min, function(err, list) {
```

Comment:
Ow ok... I was worried, thinking there was a bug. As it is a simple example, if the main data provider fail, there is no why to continue. So throw it.